src/structures/GuildMember.js

```diff
@@ -248,7 +248,9 @@ class GuildMember extends Base {
    */
   edit(data, reason) {
     if (data.channel) {
-      data.channel_id = this.client.channels.resolve(data.channel).id;
+      data.channel = this.client.channels.resolve(data.channel);
```

Comment:
You could use `GuildChannelStore#resolve` to restrict the valid channels further down to just channels in the guild.
```suggestion
      data.channel = this.guild.channels.resolve(data.channel);
```